src/protocols/request-for-quote-erc20.ts

```diff
@@ -0,0 +1,119 @@
+'use strict'
+import {
+  toDecimalString,
+  createOrderERC20,
+  createOrderERC20Signature,
+  toAtomicString,
+  getCostFromPricing,
+} from '@airswap/utils'
+import { Protocols } from '@airswap/constants'
+
+import { Protocol } from './protocol'
+import { decimals, result, error } from '../utils'
+
+export class RequestForQuoteERC20 extends Protocol {
+  constructor(config: any) {
+    super(config, Protocols.RequestForQuoteERC20)
+  }
+
+  async received(
+    id: any,
+    method: any,
+    params: any,
+    respond: any,
+    ws: WebSocket
+  ) {
+    if (
+      method === 'getSignerSideOrderERC20' ||
+      method === 'getSenderSideOrderERC20'
+    ) {
+      if (Number(params.chainId) !== this.config.chainId) {
+        respond(
+          error(id, {
+            code: -33601,
+            message: 'Not serving chain',
+          })
+        )
+        return
+      }
+
+      let { signerToken, senderWallet, senderToken, swapContract } = params
+      let signerAmount
+      let senderAmount
+
+      const signerDecimals = decimals[signerToken.toLowerCase()]
```

Comment:
signerToken or senderToken could be undefined here, that would stop the process, maybe fast fail to catch this behavior ?